### ISSUE TITLE
Change to Set argv parsing behavior in 5.x?

### ISSUE THREAD

**Issue by MSeal** [ipython/traitlets#625]
I didn't see an existing issue for this yet, but with traitlets 5 release we had this reported in nbconvert: https://github.com/jupyter/nbconvert/issues/1410 related to set assignment from command line not working with the upgrade. Specifically `--TagRemovePreprocessor.remove_cell_tags={\"noeffect\"}` works with trailets 4 but not 5, though `[\"noeffect\"]` works in both for `remove_cell_tags = Set(Unicode())`.

Locally there isn't a problem with assigning from a set or a list inside a python process against a `Set` traitlet attribute. Tracing the issue further I got to https://github.com/jupyter/jupyter_core/blob/master/jupyter_core/application.py#L226 and it seems like the issue is in argv parsing changing behavior here.

Is there a new pattern expected for parse_command_line parsing for `Set` traits or now or is this a bug with this particular pattern?

**Comment by minrk** [comment]
Yeah, this is the same as ~all the reported 5.x issues so far. Traitlets 5.x has actual support for containers on the command-line, e.g.

```
--TagRemovePreprocessor.remove_cell_tags noeffect \
--TagRemovePreprocessor.remove_cell_tags othertag
```

without needing any brackets, double-quotes, etc.

As a result, we've deprecated the problematic literal-eval that was the only way to do it in 4.x. The deprecated literal-eval is *supposed* to still work, but it is a lot more complicated to keep it working and there are a million possible cases, so various cases needed updating after 5.x (#608, #612, #614). This is another case of the same thing, where the deprecation handling checked for `[]` but not `{}`.

**Comment by minrk** [comment]
fixed by #629

**Comment by MSeal** [comment]
Understood, thanks for getting a fix made!
